squad-create-skipfile-reproducers

```diff
@@ -0,0 +1,237 @@
+                cleaned_reason["tests"] = [reason["tests"]]
+
+            cleaned_reason["projects"] = projects
+
+            reason_list.append(cleaned_reason)
+
+    tmp_reproducer_filename = "skipfile-reproducer"
+
+    for project in [look_up_project_by_branch(project) for project in args.branches]:
+        reproducer_script_name = f"skipfile-reproducer-{args.group}-{project}"
+        if pathlib.Path(reproducer_script_name).exists():
+            pathlib.Path.unlink(pathlib.Path(reproducer_script_name))
+        tmp_custom_reproducer_filename = reproducer_script_name + "_tmp_reproducer"
+        for device in args.device_names:
+            try:
+                tmp_reproducer_filename = squadutilslib.get_reproducer(
+                    args.group,
+                    project,
+                    device,
+                    args.debug,
+                    args.build_names,
+                    args.suite_name,
+                    args.search_build_count,
+                    tmp_reproducer_filename,
+                    args.allow_unfinished,
+                    local=True,
+                )
+            except squadutilslib.ReproducerNotFound:
+                logger.error(
+                    f"No reproducer could be found for {args.group} {project} {device} {args.build_names}"
+                )
+                return -1
+
+            for reason in reason_list:
+                if project in reason["projects"]:
+                    custom_command = squadutilslib.create_ltp_custom_command(
+                        tests=reason["tests"]
+                    )
+                    reproducer_file = squadutilslib.create_custom_reproducer(
+                        tmp_reproducer_filename,
+                        args.suite_name,
+                        custom_command,
+                        local=True,
+                        filename=tmp_custom_reproducer_filename,
+                    )
+                    reproducer_scripts.append(reproducer_script_name)
+                    with open(reproducer_script_name, "a+") as multiple_reproducer_file:
+                        reproducer = (
+                            pathlib.Path(reproducer_file)
+                            .read_text(encoding="utf-8")
+                            .split("\n")
+                        )
+                        for line in reproducer:
+                            # Don't write back the #!/bin/bash part of reproducer
+                            if "#!/bin/bash" != line.strip():
+                                multiple_reproducer_file.write(line + "\n")
+
+                    logger.debug(reproducer_file)
+
+        if pathlib.Path(tmp_custom_reproducer_filename).exists():
+            pathlib.Path.unlink(pathlib.Path(tmp_custom_reproducer_filename))
+        if pathlib.Path(tmp_reproducer_filename).exists():
+            pathlib.Path.unlink(pathlib.Path(tmp_reproducer_filename))
+
+    logger.info("Finished creating skipfile reproducers.")
```

Comment:
should probably print the filename here so we know.

Reply:
Done